src/pages/development/components/app-server.md

```diff
@@ -0,0 +1,196 @@
+  
+GraphQlStateTest is a test that looks for state changes in service objects created by ObjectManager. It does this by running a GraphQL query twice, and comparing the state of the service objects before and after the second query.
+
+There are two files used to filter and skip classes that are safe to be reused in a stateful application:
+
+- `dev/tests/integration/testsuite/Magento/GraphQl/_files/state-skip-list.php`—Skips the comparison of objects by their class name or virtual name (as defined in the `di.xml` file).
+- `dev/tests/integration/testsuite/Magento/GraphQl/_files/state-filter-list.php`—Filters out that properties that should be compared.
+
+<InlineAlert variant="info" slots="text" />
+
+There are similar files in the `magento/magento2ee` GitHub repository for classes that are specific to `magento2ee`. Other repositories can use their own files in a similar way.
+
+In the `state-filter-list.php` file, there are three different types of filtering:
+  
+- The `all` section filters these property names from all of the service objects.
+- The `parents` section first compares the service object to the specified class or interface using the `instanceof` operator. If it returns true, then those filters are applied.
+- The `services` section checks for direct matches of the `serviceName`. The `serviceName` can either be the class name, or virtual type of preferences (as defined in `di.xml` files) before applying the properties filter.
+
+If you are working on a failure and it does not look like it is safe to add to the skip or filter list, then consider if you can refactor the class in a [backwards-compatible](https://developer.adobe.com/commerce/contributor/guides/code-contributions/backward-compatibility-policy/) way to use Factories of the service classes that have mutable state. If it is the class itself that has mutable state, then try to rewrite it in a way to avoid mutable state. If the mutable state is required for performance reasons, then implement `ResetAfterRequestInterface` and use `_resetState()` to reset the object back to its initial constructed state.
+
+If the class is failing because of a `Typed property $x must not be accessed before initialization` error, then the problem is because the property is not initialized by the constructor. This is a form of temporal coupling, because the object cannot be used after it is initially constructed. This happens even if the property is private because the Collector that gets the data from the properties using PHP's reflection feature. In this case, one of the best things to do is to refactor the class to avoid temporal coupling, and also to avoid mutable state. If that does not work, the property can change its type to a nullable type and be initialized to null. Or if the property is an array, it may be okay to initialize the property as an empty array.
```

Comment:
```suggestion
If the class is failing because of a `Typed property $x must not be accessed before initialization` error, then the property is not initialized by the constructor. This is a form of temporal coupling, because the object cannot be used after it is initially constructed. This happens even if the property is private because the Collector gets the data from the properties using PHP's reflection feature. In this case, refactor the class to avoid temporal coupling, and also to avoid mutable state. If that does not work, the property can change its type to a nullable type and be initialized to null. If the property is an array, it may be okay to initialize the property as an empty array.
```